msf/output.go

```diff
@@ -170,6 +170,26 @@ func _paletteOutput(
 		}
 
 		if line.IsLinearOrArcMove() {
+			// check if this is an unretract after a tool change and adjust the E value
+			if state.ToolChangeJustSeen && line.IsUnretract() && state.ToolChangeCount > 1 {
+				var postToolChangeUnretract float32
+				if palette.Type == TypeElement && state.Palette.MaterialMeta[state.CurrentTool].DensifierRetractDistance > 0 {
+					// use material's densifier's retract length for element
+					postToolChangeUnretract = state.Palette.MaterialMeta[state.CurrentTool].DensifierRetractDistance
+				} else {
+					// unretract by project's retract length
+					postToolChangeUnretract = state.Palette.RetractDistance[state.CurrentTool]
+				}
+				// replace only the E value in the command
+				eParam := line.Params["e"]
+				lineToWrite := strings.ReplaceAll(line.Raw, fmt.Sprintf("E%g", eParam), fmt.Sprintf("E%g", postToolChangeUnretract))
```

Comment:
Can we use `gcode.FormatFloat(postToolChangeUnretract)` to ensure consistent float formatting?

Alternatively, should we just construct a new `gcode.Command` rather than doing manual string replacement? See the `msf.getRestart` function, might be simpler to just use that